cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -58,6 +59,9 @@ func CreateInitKubeConfigFiles(masterEndpoint, pkiDir, outDir string) error {
 	if err != nil {
 		return err
 	}
+	// Lowercasing here is needed, because Node names in kubernetes are lower-case, and the Node Admission plugin strips "system:node:" from the CN below
+	// and compares it strictly to the (kubernetes lower-cased) Node name.
+	hostname := strings.ToLower(hostname)
```

Comment:
currently, kubeadm skips the cert generation step if the cert already exists. if kubeadm wants to support upgrades, it will now need to read an existing client cert and verify it is valid (not expired and subject matches what it has been asked to generate), and regenerate the cert if needed